R/Connection.R

```diff
@@ -647,16 +717,90 @@ odbcListDrivers <- function(keep = getOption("odbc.drivers_keep"), filter = getO
 
 #' List Available Data Source Names
 #'
-#' List the available data sources on your system. See the [DSN Configuration
-#' files](https://github.com/r-dbi/odbc#dsn-configuration-files) section of the
-#' package README for details on how to install data sources for the most
-#' common databases.
+#' @description
+#'
+#' Collect information about the available data source names (DSNs).
 #'
-#' @return A data frame with two columns.
+#' @return A data frame with two columns:
 #' \describe{
-#'   \item{name}{Name of the data source}
-#'   \item{description}{Data Source description}
+#'   \item{name}{Name of the data source. The entries in this column can be
+#'     passed to the `dsn` argument of [dbConnect()].}
+#'   \item{description}{Data source description.}
 #' }
+#'
+#' @section Configuration:
+#'
+#' This function interfaces with the driver manager to collect information
+#' about the available data source names.
+#'
+#' For **MacOS and Linux**, the odbc package supports the unixODBC driver
+#' manager. unixODBC looks to the `odbc.ini` _configuration file_ for information
+#' on DSNs. Install unixODBC with `brew install unixodbc`, and find the
+#' location(s) of your `odbc.ini` file(s) with `odbcinst -j`.
+#'
+#' In this example `odbc.ini` file:
+#'
+#' ```
+#' [MySQL]
+#' Driver = MySQL Driver
+#' Database = test
+#' Server = 127.0.0.1
+#' User = root
+#' password = root
+#' Port = 3306
+#' ```
+#'
+#' ...the data source name is `MySQL`, which will appear in the `name`
+#' column of this function's output. To pass the DSN as the `dsn` argument to
+#' [dbConnect()], pass it as a string, like `"MySQL"`.
```

Comment:
Again, I think it would be useful to help people build the mental model that this is just a lookup table for supplying default arguments. And it also uses the driver lookup table.

Reply:
[d6a6da0](https://github.com/r-dbi/odbc/pull/649/commits/d6a6da062e4a1156cd7f29f8c23c10aa9e054a63).